extensions/cornerstone-dicom-pmap/src/getSopClassHandlerModule.ts

```diff
@@ -0,0 +1,224 @@
+    SOPInstanceUID,
+    SeriesDescription,
+    SeriesNumber,
+    SeriesDate,
+    SOPClassUID,
+    wadoRoot,
+    wadoUri,
+    wadoUriRoot,
+  } = instance;
+
+  const displaySet = {
+    // Parametric map use to have the same modality as its referenced volume
+    // but "other" is used here just to remove its thumbnail on the Study Browser
+    // without making any extra change to the viewer. It also makes more sense
+    // to do not see a thumbnail for an OT series other than CT or MR ones.
+    Modality: 'OT',
+    isReconstructable: true, // by default for now
+    displaySetInstanceUID: `pmap.${utils.guid()}`,
+    SeriesDescription,
+    SeriesNumber,
+    SeriesDate,
+    SOPInstanceUID,
+    SeriesInstanceUID,
+    StudyInstanceUID,
+    SOPClassHandlerId,
+    SOPClassUID,
+    referencedImages: null,
+    referencedSeriesInstanceUID: null,
+    referencedDisplaySetInstanceUID: null,
+    referencedVolumeURI: null,
+    referencedVolumeId: null,
+    isDerivedDisplaySet: true,
+    loadStatus: {
+      loading: false,
+      loaded: false,
+    },
+    sopClassUids,
+    instance,
+    instances: [instance],
+    wadoRoot,
+    wadoUriRoot,
+    wadoUri,
+    hasVolumeData: true,
+  };
+
+  const referencedSeriesSequence = instance.ReferencedSeriesSequence;
+
+  if (!referencedSeriesSequence) {
+    console.error('ReferencedSeriesSequence is missing for the parametric map');
+    return;
+  }
+
+  const referencedSeries = referencedSeriesSequence[0] || referencedSeriesSequence;
+
+  displaySet.referencedImages = instance.ReferencedSeriesSequence.ReferencedInstanceSequence;
+  displaySet.referencedSeriesInstanceUID = referencedSeries.SeriesInstanceUID;
+
+  displaySet.getReferenceDisplaySet = () => {
+    const { displaySetService } = servicesManager.services;
+
+    if (displaySet.referencedDisplaySetInstanceUID) {
+      return displaySetService.getDisplaySetByUID(displaySet.referencedDisplaySetInstanceUID);
+    }
+
+    const referencedDisplaySets = displaySetService.getDisplaySetsForSeries(
+      displaySet.referencedSeriesInstanceUID
+    );
+
+    if (!referencedDisplaySets || referencedDisplaySets.length === 0) {
+      throw new Error('Referenced displaySet is missing for the parametric map');
+    }
+
+    const referencedDisplaySet = referencedDisplaySets[0];
+
+    displaySet.referencedDisplaySetInstanceUID = referencedDisplaySet.displaySetInstanceUID;
+
+    return referencedDisplaySet;
+  };
+
+  displaySet.getReferencedVolumeId = () => {
+    if (displaySet.referencedVolumeId) {
+      return displaySet.referencedVolumeId;
+    }
+
+    const referencedDisplaySet = displaySet.getReferenceDisplaySet();
+    const referencedVolumeURI = referencedDisplaySet.displaySetInstanceUID;
+    const referencedVolumeId = `${VOLUME_LOADER_SCHEME}:${referencedVolumeURI}`;
+
+    displaySet.referencedVolumeURI = referencedVolumeURI;
+    displaySet.referencedVolumeId = referencedVolumeId;
+
+    return referencedVolumeId;
+  };
```

Comment:
There's nothing asynchronous about this. I'm wondering why we're creating a method for it. Does it need to be a method, or can we simply derive the values?